src/components/FilterPanel.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core";
+import { useTheme } from '@material-ui/core/styles';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Switch from "@material-ui/core/Switch";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Container from "@material-ui/core/Container";
+
+
+const useStyles = makeStyles((theme) => ({
+    formControlLabel: {
+        "text-align": "left",
+        "align-items": "start",
+        "margin-top": theme.spacing(1),
+        "width": `calc(100% - ${theme.spacing(4)}px)`
+    },
+    mdPanel: {
+        "position": "sticky",
+        "top": 0,
+        "height": "200px"
+    }
+}));
+
+export default function FilterPanel(props) {
+    const classes = useStyles();
+    const theme = useTheme();
+    const mdSize = useMediaQuery(theme.breakpoints.up('md'));
+
+    const [hasAppointments, setHasAppointments] = useState(true);
+    const [vaxType, setVaxType] = useState('Any');
+
+    const { data, onChange } = props;
+
+    useEffect(() => {
+        onChange({
+            hasAppointments: hasAppointments ?
+                d => !!d.hasAppointments === hasAppointments
+                :
+                () => true,
+            vaxType: vaxType !== 'Any' ?
```

Comment:
Same idea here, I think the ternary is slightly harder to understand and could be replaced with `vaxType: d => vaxType === 'Any' || (d.extraData && d.extraData['Vaccinations offered'] && d.extraData['Vaccinations offered'].includes(vaxType))`.

I guess it comes down to preference, maybe the one-liner is harder to read, in which case ignore me!

Reply:
To me this feels a little less transparent, but if others prefer this way too I could change it.

Reply:
I think pretty much any use of the ternary operator that isn't for short simple expressions should be avoided, and it definitely shouldn't be combined with arrow functions. I would move the test into the function, and further simplify a bit:

```js
d => {
  if (vaxType !== 'Any') {
    let vaxOffered = d.extraData && d.extraData['Vaccinations offered'];
    return vaxOffered && vaxOffered.includes(vaxType);
  } else {
    return true;
  }
```

(untested, but I think that's equivalent)

Reply:
Just as you have an aversion to ternary operators, I have one for assignment expressions with `&&` or `||`. They always require a second look. As a compromise, how about we go fully explicit?

```js
           vaxType: d => {
                if (vaxType === 'Any') {
                    return true;
                } else {
                    if (d.extraData && d.extraData['Vaccinations offered']) {
                        return d.extraData['Vaccinations offered'].includes(vaxType);
                    } else {
                        return false;
                    }
                }
            }
```

Reply:
Err, to be clear, the concern is not the ternary operator — it's the ternary expression being split over 9 lines that include an arrow function, where the test is not obviously better outside the arrow function than in.

Why is `&&` ok in
```js
if (d.extraData && d.extraData['Vaccinations offered']) {
```
but not in
```js
return vaxOffered && vaxOffered.includes(vaxType);
```
?

Also, part of the point of the simplification (and introduction of `let vaxOffered`) was to avoid searching the `d.extraData` dictionary twice (and cut down line length and cognitive load). Yours undoes that.

If you really like `?:` and dislike `&&` I'd replace
```js
    return vaxOffered && vaxOffered.includes(vaxType);
```
with
```js
    return vaxOffered ? vaxOffered.includes(vaxType) : false;
```
which, although silly to my eye, would seem better than having an entire 5-line if/else block to accomplish the same thing.

But I am In No Way King, and your revised solution eliminates the mass confusion point of the long ternary-operator-with-arrow-function-within, so is definitely an improvement over the original.

Reply:
I'm not trying to pick a fight here, man--just addressing readability. I'll go with the most recent revision with the explicit if-else's so we're not running in circles.